src/source/worker.ts

```diff
@@ -85,126 +85,134 @@ export default class Worker {
-    }
+        this.actor.registerMessageHandler('reloadTile', (mapId: string, params: WorkerTileParameters) => {
+            return this._getWorkerSource(mapId, params.type, params.source).reloadTile(params);
+        });
 
-    removeTile(mapId: string, params: TileParameters & {
-        type: string;
-    }, callback: WorkerTileCallback) {
-        this.getWorkerSource(mapId, params.type, params.source).removeTile(params, callback);
-    }
+        this.actor.registerMessageHandler('abortTile', (mapId: string, params: TileParameters) => {
+            return this._getWorkerSource(mapId, params.type, params.source).abortTile(params);
+        });
 
-    removeDEMTile(mapId: string, params: TileParameters) {
-        this.getDEMWorkerSource(mapId, params.source).removeTile(params);
-    }
+        this.actor.registerMessageHandler('removeTile', (mapId: string, params: TileParameters) => {
+            return this._getWorkerSource(mapId, params.type, params.source).removeTile(params);
+        });
 
-    removeSource(mapId: string, params: {
-        source: string;
-    } & {
-        type: string;
-    }, callback: WorkerTileCallback) {
+        this.actor.registerMessageHandler('removeSource', async (mapId: string, params: RemoveSourceParams) => {
+            if (!this.workerSources[mapId] ||
+                !this.workerSources[mapId][params.type] ||
+                !this.workerSources[mapId][params.type][params.source]) {
+                return;
+            }
 
-        if (!this.workerSources[mapId] ||
-            !this.workerSources[mapId][params.type] ||
-            !this.workerSources[mapId][params.type][params.source]) {
-            return;
-        }
+            const worker = this.workerSources[mapId][params.type][params.source];
+            delete this.workerSources[mapId][params.type][params.source];
 
-        const worker = this.workerSources[mapId][params.type][params.source];
-        delete this.workerSources[mapId][params.type][params.source];
+            if (worker.removeSource !== undefined) {
+                worker.removeSource(params);
+            }
+        });
+
+        this.actor.registerMessageHandler('setReferrer', async (_mapId: string, params: string) => {
+            this.referrer = params;
+        });
+
+        this.actor.registerMessageHandler('syncRTLPluginState', (mapId: string, params: PluginState) => {
+            return this._syncRTLPluginState(mapId, params);
+        });
+
+        this.actor.registerMessageHandler('loadWorkerSource', async (_mapId: string, params: string) => {
+            return new Promise((resolve, reject) => {
+                try {
+                    this.self.importScripts(params);
+                    resolve();
+                } catch (ex) {
+                    // This is done since some error messages are not serializable
+                    reject(ex.toString());
+                }
+            });
+        });
```

Comment:
I think in general we should avoid constructing new Promises especially now that we have async functions, so this could become:

```suggestion
        this.actor.registerMessageHandler('loadWorkerSource', async (_mapId: string, params: string) => {
            try {
                this.self.importScripts(params);
            } catch (ex) {
                // This is done since some error messages are not serializable
                throw ex.toString();
            }
        });
```